osu.Game/Online/DownloadTrackingComposite.cs

```diff
@@ -127,30 +144,42 @@ private void attachDownload(ArchiveDownloadRequest<TModel> request)
 
         private void onRequestSuccess(string _) => Schedule(() => State.Value = DownloadState.Importing);
 
-        private void onRequestProgress(float progress) => Schedule(() => Progress.Value = progress);
+        private void onRequestProgress(double progress) => Schedule(() => Progress.Value = progress);
 
         private void onRequestFailure(Exception e) => Schedule(() => attachDownload(null));
 
         private void itemUpdated(ValueChangedEvent<WeakReference<TModel>> weakItem)
         {
             if (weakItem.NewValue.TryGetTarget(out var item))
-                setDownloadStateFromManager(item, DownloadState.LocallyAvailable);
+            {
+                Schedule(() =>
+                {
+                    if (!item.Equals(Model.Value))
+                        return;
+
+                    if (!VerifyDatabasedModel(item))
+                    {
+                        State.Value = DownloadState.NotDownloaded;
+                        return;
+                    }
```

Comment:
so this will silently fail and possibly keep importing copies of the same map with a bad checksum, with no indication to the user as to what's going wrong? sounds not ideal

Reply:
that's how the multiplayer's download button behaves, but yeah I don't like the UX there either, logging error will do at least, not sure what the message should be other than a `the imported beatmap does not match online version`, any ideas?

Reply:
sounds tolerable for a start, certainly better than nothing